presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/BaseJdbcClient.java

```diff
@@ -305,7 +306,166 @@ protected static ResultSet getColumns(JdbcTableHandle tableHandle, DatabaseMetaD
     @Override
     public ConnectorSplitSource getSplits(JdbcIdentity identity, JdbcTableHandle tableHandle)
     {
-        return new FixedSplitSource(ImmutableList.of(new JdbcSplit(Optional.empty())));
+        List<JdbcSplit.RangeInfo> ranges = null;
+        try (Connection connection = connectionFactory.openConnection(identity)) {
+            DatabaseMetaData metaData = connection.getMetaData();
+            // determine a column to split the whole table scan
+            ColumnInfo splitColumn = getSplitColumn(metaData, tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName());
+
+            if (splitColumn != null) {
+                log.info("Table: %s.%s.%s, splitColumn: %s", tableHandle.getCatalogName(), tableHandle.getSchemaName(), tableHandle.getTableName(), splitColumn);
+
+                // collect the min/max stats for the specified split column
+                String minMaxSql = format("select min(%s) AS MIN_VALUE, max(%s) AS MAX_VALUE from %s",
```

Comment:
This will do a full table scan in the underlying RDBMS, if `splitColumn.getName()` is not part of an index.
This will do a full index scan, if `splitColumn.getName()` is not a first column of some index.

This is another reason why you need to choose partitioning column wisely.